### ISSUE TITLE
[Bug]: Door lock does not show the correct status in Home app

### ISSUE THREAD

**Issue by phamhainguyen68** [homebridge-plugins/homebridge-eufy-security#110]
### What happened?

Hello! First, I would like to show my appreciation to you for creating and maintaining a wonderful plugin for the community.

And now is the fun part: I found a bug: **Eufy Door lock does not show the correct status in Home app**. It always show as Unlocked. 

If I try to lock it from the Home app, it just keeps spinning without making any progress or status update.

If I manually (or through Eufy app) unlock the lock, I can then use the Home app to lock it. Home app then shows the correct status (Locked)for only 30 seconds or so, and it updates the status back to Unlocked.

I have tried to restart the lock, its wifi bridge, as well as recalibrate the lock from Eufy app.

I am on the latest beta version 2.1.0 beta 8. But I tried with the official 2.0.1 to no avail. I also tried uninstalling the plug-in, clear all the Homebridge cache and start over. None of them addresses the issue.

The lock model is T8510 (https://us.eufylife.com/products/t8510). It comes with a wifi bridge T8021 to connect itself to our wifi network (since the lock is bluetooth-only).

### Device Type

other (UnSupported)

### Plugin Version

beta (Best Effort)

### HomeBridge Version

v1.4.x (Supported)

### NodeJS Version

v16 (Supported)

### Operating System

Ubuntu 18.04+ (Supported)

### Relevant log output

```shell
Log from Homebridge: 
[7/13/2022, 3:32:49 PM] [EufySecurity-2.1.0-beta.8] DEBUG: Front Door LockStatus false
[7/13/2022, 3:32:49 PM] [EufySecurity-2.1.0-beta.8] DEBUG: Front Door Triggered GET LockCurrentState 0
[7/13/2022, 3:32:49 PM] [EufySecurity-2.1.0-beta.8] DEBUG: Front Door LockStatus false
[7/13/2022, 3:32:49 PM] [EufySecurity-2.1.0-beta.8] DEBUG: Front Door Triggered GET LockTargetState 0
```

**Comment by thieren** [comment]
Hi @phamhainguyen68 

as mentioned on discord I am currently trying to get this working with some other users who help me debug this. You're welcome to join us.

**Comment by phamhainguyen68** [comment]
Thank you, @thieren, for resolving this issue. Below is the summary of the error (quoted from thieren on Discord):

```
The main issue were some bugs in eufy-security-client. this library handles the lock with two parameters: DeviceLocked (true/false) and DeviceLockState (a number value 1-7). The thing was that this was muddled up and in specific cases the DeviceLockState was handled as boolean. That resulted in false values returned from the library.

I'll open a PR for the library and I'm confident that this will be updated in the next release.

The last issue you've reported was a missing update in the plugin itself. So in summary both plugin and library had to be changed.
```

**Comment by thieren** [comment]
I'll keep this open, since this can only be really fixed after this [PR](https://github.com/bropat/eufy-security-client/pull/184) in eufy-security-client is merged.

**Comment by thieren** [comment]
Hi @phamhainguyen68 

I updated to the latest eufy-security-client version with 2.1.0-beta.14

The door lock states should be fine now.

**Comment by thieren** [comment]
Is fixed with release of 2.1.0
